### ISSUE TITLE
Performance visualization page

### ISSUE THREAD

**Issue by rth** [pyodide/pyodide#2466]
@wlach mentioned on gitter,
> I'm looking at the benchmarking results for pandas right now. I'm curious, has there been any thought about collecting these results on the main branch into a single data store? I'm thinking of a simple github pages site as an MVP

I was also thinking that having an HTML page which displays historical data on performance and code sizes would be great.

It could be as simple as,
 - an HTML page at https://github.com/pyodide/pyodide-blog which does these plots using pyodide, and data is loaded from some json that we update say for each release
 - or indeed a specific application, that maybe monitors performance for each commit. Though then it's a bit more complex, and maybe it would make more sense to re-use https://asv.readthedocs.io/en/stable/using.html

Note that we are also planning to completely change the benchmark suite, to get closer to the used by CPython and or scientific python projects (e.g. pandas, numpy, and scikit-learn) already have test suites, and we could take a few representative cases from there.

**Comment by wlach** [comment]
asv looks fantastic! It looks like it has setup methods which could be used to initialize selenium or set up tests: 

https://asv.readthedocs.io/en/stable/writing_benchmarks.html#writing-benchmarks

**Comment by mdboom** [comment]
Yeah, `asv` could be useful here, but I'm biased.

For some inspiration (though I don't know whether it makes sense to use directly), see speed.python.org, which is based on [codespeed](https://github.com/tobami/codespeed).  These haven't seen much maintenance in a while, but may get revived.

**Comment by wlach** [comment]
One thing asv seems pretty hardcore about specifying is the machine on which the benchmarks are run, which brings up an issue with any benchmarking *we* might want to do: performance on a VPS host is notoriously variable (even taking aside the fact that they might randomly upgrade the hardware on us). Although it looks like the [scipy people had some success with using GitHub actions to benchmark their stuff](https://labs.quansight.org/blog/2021/08/github-actions-benchmarks/), so maybe it's possible.

**Comment by rth** [comment]
>  the CPython performance team is still using the "machine sitting under someone's desk" model.

Yeah, AFAIK https://pandas.pydata.org/speed/ which uses ASV also had that setup in the past. I have a desktop under my desk that could be used for such purposes, but unfortunately my cat sometimes walks on the power button, and then it switches off. So that's not ideal ^^ 

> Maybe bare metal cloud options could work, but that's expensive.

Bare metal servers from https://www.kimsufi.com/us/en/servers.xml could be rented for as little as 240$ per year for somewhat outdated hardware (e.g. 4CPU  Xeon E3-1225v2 from 2012, 16GB RAM). That's plenty for benchmarks. We can probably get one from Pyodide funds if anyone is willing to setup ASV there.
